sim/battle.ts

```diff
@@ -2228,11 +2228,12 @@ export class Battle {
 		if (this.activePerHalf > 2) {
 			if (move.target === 'adjacentFoe' || move.target === 'normal' || move.target === 'randomNormal') {
 				// even if a move can target an ally, auto-resolution will never make it target an ally
-				// i.e. if both your opponents faint before you use Flamethrower, it will fail instead of targeting your all
+				// i.e. if both your opponents faint before you use Flamethrower, it will fail instead of targeting your ally
 				const adjacentFoes = pokemon.adjacentFoes();
-				if (adjacentFoes.length) return this.sample(adjacentFoes);
-				// no valid target at all, return a possibly-fainted foe for any possible redirection
```

Comment:
I don't understand why you'd want to remove this comment. On the contrary, it should be expanded upon to prevent this issue from happening again.